### PR TITLE
fixed error-handling in manager-services, implemented button-extensions as extension

### PR DESCRIPTION
**Pull request by nnatter**
this pr fixes the wrong usage of the jquery deferred object in the manager-services of the media- and contact-bundle.
also bundle specific buttons are implemented as a real extension now and the not usable permission-tab in media-bundle is removed.

__tasks:__

- [x] fixed error-handling in manager-services
- [x] implemented bundle specific buttons as extension
- [x] removed permission-tab in media-bundle
- [ ] test coverage
- [x] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js [hunk 1]
```diff
@@ -250,8 +258,10 @@ define([
                 requests.push(removeAccountContact(accountId, id));
             }.bind(this));
 
-            $.when.apply(null, requests).then(function() {
+            $.when.apply(null, requests).done(function() {
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js]: why did you change that?

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/services/contact-manager.js [hunk 2]
```diff
@@ -14,15 +14,13 @@ define([
...
-], function(
-    Util,
-    Mediator,
-    Contact,
-    Title,
-    Position,
-    Category
-) {
+    'sulumedia/models/media'
+], function(Util,
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/services/contact-manager.js]: format wrong

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-create-overlay/main.js [hunk 3]
```diff
@@ -95,28 +95,32 @@ define(['services/sulumedia/collection-manager'], function(CollectionManager) {
...
         /**
+         * Creates a new Collections with the form-data of the overlay
+         * @returns {Boolean} returns false if form-data was not valid
+         */
+        addCollection: function() {
+            var collection = this.sandbox.form.getData(constants.newFormSelector);
+            collection.parent = this.options.parent;
+
+            CollectionManager.save(collection).done(function(collection) {
+                Mediator.emit('sulu.media.collection-create.created', collection);
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-create-overlay/main.js]: here you should use `this.sandbox.emit()`

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js (hunk 1 above)
Comment (nnatter) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js]: @wachterjohannes this is the proper way to wait for a deferred-resolve. (see: https://api.jquery.com/category/deferred-object/)

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-create-overlay/main.js (hunk 3 above)
Comment (nnatter) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-create-overlay/main.js]: @wachterjohannes the sandbox is already closed when this is executed (see line 104)

Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-create-overlay/main.js]: hmm ok ...
